Framework/PythonInterface/mantid/plots/mantidimage.py

```diff
@@ -0,0 +1,93 @@
+THRESHOLD = 100
+
+
+class ImageIntensity(Enum):
+    LIGHT = 1
+    DARK = 2
+
+
+class MantidImage(mimage.AxesImage):
+    def __init__(self, ax,
+                 cmap=None,
+                 norm=None,
+                 interpolation=None,
+                 origin=None,
+                 extent=None,
+                 filternorm=1,
+                 filterrad=4.0,
+                 resample=False,
+                 **kwargs):
+        super().__init__(ax,
+                         cmap=cmap,
+                         norm=norm,
+                         interpolation=interpolation,
+                         origin=origin,
+                         extent=extent,
+                         filternorm=filternorm,
+                         filterrad=filterrad,
+                         resample=resample,
+                         **kwargs)
+
+        # Increase the default pen thickness
+        self.update_pen_thickness(1.5)
+
+    def draw(self, renderer, *args, **kwargs):
+        self.update_pen_color()
+        super().draw(renderer, *args, **kwargs)
+
+    def update_pen_color(self, color=None):
+        """Update the pen color used to draw tool in the matplotlib toolbar, e.g
+        the zoombox. If no color is specified the color is automatically determined
+        by considering how dark, or light the image is and setting a pen appropriately.
+        If the canvas is not a MantidFigureCanvas, the method will be skipped.
+        :param color: A qt color instance
+        """
+        from workbench.plotting.mantidfigurecanvas import MantidFigureCanvas
+        if not isinstance(self.axes.get_figure().canvas, MantidFigureCanvas):
+            return
+        if color is None:
+            image_intensity = self._calculate_greyscale_intensity()
+            if image_intensity == ImageIntensity.DARK:
+                color = Qt.white
+            else:
+                color = Qt.black
+        self.axes.get_figure().canvas.pen_color = color
+
+    def update_pen_thickness(self, value):
+        """Update the pen thickness used to draw tool in the matplotlib toolbar, e.g
+        the zoombox.
+        If the canvas is not a MantidFigureCanvas, the method will be skipped.
+        :param value: Thickness of the pen line
+        """
+        from workbench.plotting.mantidfigurecanvas import MantidFigureCanvas
+        if isinstance(self.axes.get_figure().canvas, MantidFigureCanvas):
+            self.axes.get_figure().canvas.pen_thickness = value
+
+    def _calculate_greyscale_intensity(self) -> ImageIntensity:
+        """
+        Calculate the intensity of the image in greyscale.
+        The intensity is given in the range [0, 255] where:
+        -0 is black - i.e. a dark image and
+        -255 is white, a light image.
+        """
+        rgb = self.to_rgba(self._A, alpha=None, bytes=True, norm=True)
+        r, g, b = rgb[:, :, 0], rgb[:, :, 1], rgb[:, :, 2]
+        grey = 0.2989 * r + 0.5870 * g + 0.1140 * b
```

Comment:
Maybe worth adding a comment here as to where you got these numbers to multiply r, g and b by to give the greyscale intensity? Also is the Max value of grey 255? If so then should the threshold be 100 or 127.5??

Reply:
I've added a comment where they were taken from. Yeah the max value would be 255, so half would be 127.5, but that doesn't necessarily mean its halfway between light and dark. Its fairly qualitative in terms of what constitutes "dark" and "light", to me at 127 a few too many images were being classified as dark, 100 felt like a good number - but I guess others would have different interpretations.

![image](https://user-images.githubusercontent.com/56431339/100341142-15c90580-2fd4-11eb-9e46-4e44c3a0a164.png)